toolkit/resources/imageconfigs/additionalfiles/iso_initrd/root/runliveinstaller

```diff
@@ -83,6 +84,38 @@ else
     retry "{ mount -L $LABEL -o ro $ISO_ROOT || grep -qs $ISO_ROOT /proc/mounts ; }"
 fi
 
+# If we are running unattended (i.e., have $UNATTENDED_CONFIG_FILE), we need to make sure that we only run once.
+# We can use flock to synchronize the unattended install flows.
+if [[ -f "$UNATTENDED_CONFIG_FILE" ]]; then
+    echo "Synchronizing unattended install flows... current sessions are:"
+    who
+    thisTerminal=$(who -m | xargs | cut -d' ' -f2)
+    echo "This instance is on '$thisTerminal'"
+
+    # Generally we want to prioritize ttyS0 since that will be used for debugging. The output from the log file will
+    # contain the timestamp and color codes so it is still useful for debugging and showing liveness but it will look
+    # messy.
+    if [[ "$thisTerminal" != "ttyS0" ]]; then
+        echo "Pausing for 2 seconds to allow other session to take priority"
+        sleep 2
+    fi
+
+    # Create a lock file and open file descriptor 200 to it (200 is chosen arbitrarily)
+    # The exec will hold the file descriptor open for the duration of the script since the flock will only
+    # work on file descriptors that are open.
+    touch "$UNATTENDED_LOCK_FILE"
+    exec 200>> "$UNATTENDED_LOCK_FILE"
+
+    # If the file descriptor is already locked bail out immediately and just follow the other installer's log. The tail
+    # command will follow the log file and should never exit. We must ensure that this instance of the installer does
+    # not run any further so we always exit 0 just in case.
+    flock -x -n 200 || \
+        { echo "Unattended install already running, following other installer's log instead" ; \
+        sleep 5 ; \
+        tail -f -n 10000 /installer/log.txt ; \
+        exit 0 ; }
```

Comment:
Refactoring suggestion to achieve two goals:
- Drop the need for repeated `; \` at the end of every line.
- Be explicit about ignoring non-zero exit codes from `flock`. Thanks to that, this code should be more resilient if we ever add `set -e` to this script in the future. We tend to already do that in our scripts and I wouldn't be very surprised, if we ever wanted to introduce it here to fail fast.

```suggestion
    if ! flock -x -n 200; then
        echo "Unattended install already running, following other installer's log instead" 
        sleep 5
        tail -f -n 10000 /installer/log.txt 
        exit 0
    fi
```